File: java/pe016.java
//file: pe016.java

import java.math.BigInteger;

public class pe016 {
  public static void main( String[] args ) {
    final int limit = 1000;

    BigInteger num = BigInteger.valueOf(2);
    num = num.pow(limit);
    String numstring = num.toString();
    int ssum = 0;
    for (int i = 0; i < numstring.length(); i++)
      ssum += Character.getNumericValue(numstring.charAt(i));

    System.out.println(ssum);
  }
}

File: typescript/pe016.ts
function pe016(limit16:number): number {
  let num = BigInt(2 ** limit16)
  let string = num.toString()

  let ssum = 0
  for(let i = 0; i < string.length; i++)
    ssum += Number(string[i])
  return ssum
}

let limit16 = 1000
console.log(pe016(limit16))
